File: 1022.从根到叶的二进制数之和/1022-从根到叶的二进制数之和.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumRootToLeaf(TreeNode* root) {
        int sum = 0;
        int cur = 0;
        dfs(root, sum, cur);
        return sum;
    }

    void dfs(TreeNode* root, int& sum, int cur) {
        if( !root ) {
            return;
        }

        cur = cur*2 + root->val;
        cout << cur << endl;

        if( !root->left && !root->right ) {
            sum += cur;
            return;
        }

        dfs(root->left, sum, cur);
        dfs(root->right, sum, cur);
    }
};

File: 1022.从根到叶的二进制数之和/1022-从根到叶的二进制数之和.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumRootToLeaf(root *TreeNode) int {
    sum := 0
    cur := 0

    dfs(root, &sum, cur)
    return sum
}

func dfs(root *TreeNode, sum *int, cur int) {
    if nil == root {
        return
    }

    cur = cur*2 + root.Val

    if nil == root.Left && nil == root.Right {
        *sum = *sum + cur
        return
    }

    dfs( root.Left, sum, cur )
    dfs( root.Right, sum, cur )
}